pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -1174,13 +1174,19 @@ public ServiceUnitId getServiceUnitId(TopicName topicName) throws Exception {
     }
 
     private CompletableFuture<List<String>> getPartitionsForTopic(TopicName topicName) {
-        return pulsar.getBrokerService().fetchPartitionedTopicMetadataAsync(topicName).thenCompose(meta -> {
+        CompletableFuture<List<String>> future = new CompletableFuture<>();
+        pulsar.getBrokerService().fetchPartitionedTopicMetadataAsync(topicName).whenComplete((meta, t) -> {
+            if (t != null) {
+                future.completeExceptionally(t);
+                return;
+            }
             List<String> result = Lists.newArrayList();
             for (int i = 0; i < meta.partitions; i++) {
                 result.add(topicName.getPartition(i).toString());
             }
-            return CompletableFuture.completedFuture(result);
+            future.complete(result);
         });
+        return future;
```

Comment:
What is the difference between these changes? Why we change the `thenCompose` to `whenComplete`?